activerecord/lib/active_record/connection_adapters/schema_cache.rb

```diff
@@ -5,6 +5,14 @@
 module ActiveRecord
   module ConnectionAdapters
     class SchemaCache
+      # Override this method to support other PostgreSQL adapter.
```

Comment:
I'm guessing one of the concerns with this implementation will be the addition of postgres-specific methods and functionality into an otherwise general class.

I'll admit I'm not familiar with how the other DB adapters are initialized or whether they have equivalent type map functionality, but would it be possible to implement these changes with either an adapter-specific subclass (ex: a postgres-specific schema cache that defers to `super`) or with hook/wrapper methods where we can put all adapter-specific logic (ex: `initialize_adapter()`, `init_with_adapter()`, `clear_adapter!(), etc.)?

There are probably other better options as well, but I think an interface that's more adapter-agnostic (or at least maintains better separation of general and adapter-specific logic) will be more likely to get support.